src/solvers/refinement/bv_refinement.h

```diff
@@ -21,34 +21,46 @@ Author: Daniel Kroening, kroening@kroening.com
 class bv_refinementt:public bv_pointerst
 {
 public:
-  bv_refinementt(const namespacet &_ns, propt &_prop);
-  ~bv_refinementt();
+  struct infot
+  {
+    const namespacet *ns=nullptr;
+    propt *prop=nullptr;
+    language_uit::uit ui=language_uit::uit::PLAIN;
+    /// Max number of times we refine a formula node
+    unsigned max_node_refinement=5;
+    /// Enable array refinement
+    bool refine_arrays=true;
+    /// Enable arithmetic refinement
+    bool refine_arithmetic=true;
+  };
 
-  virtual decision_proceduret::resultt dec_solve();
+  explicit bv_refinementt(const infot &info);
 
-  virtual std::string decision_procedure_text() const
+  decision_proceduret::resultt dec_solve() override;
+
+  std::string decision_procedure_text() const override
   {
     return "refinement loop with "+prop.solver_text();
   }
 
-  // NOLINTNEXTLINE(readability/identifiers)
-  typedef bv_pointerst SUB;
-
-  // maximal number of times we refine a formula node
-  unsigned max_node_refinement;
-  // enable/disable refinements
-  bool do_array_refinement;
-  bool do_arithmetic_refinement;
+protected:
 
-  using bv_pointerst::is_in_conflict;
+  // Refine array
+  void post_process_arrays() override;
 
-  void set_ui(language_uit::uit _ui) { ui=_ui; }
+  // Refine arithmetic
+  bvt convert_mult(const exprt &expr) override;
+  bvt convert_div(const div_exprt &expr) override;
+  bvt convert_mod(const mod_exprt &expr) override;
+  bvt convert_floatbv_op(const exprt &expr) override;
 
-protected:
-  resultt prop_solve();
+  // Collect stats
```

Comment:
I don't understand the comment

Reply:
Original was: `// for collecting statistics`